### PR TITLE
Refactor and remove unnecessary in misc/pstring.cpp and dependents.

### PR DESCRIPTION
**Pull request by Nakhr11n**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

Remove unnecessary code.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: misc/pstring.cpp [hunk 1]
```diff
@@ -38,140 +38,22 @@
...
-    } else if (ch > ' ' && err >= 1) {
-      end = i;
-    }
+//    This function strips all leading and trailing spaces, keeping internal spaces. This goes for tabs too.
+int CleanupStr(char *dest, const char *src, std::size_t destlen) {
+  if ((src != nullptr) && (dest != nullptr)) {
+    std::string s{src};
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char ch) { return !std::isspace(ch); }));
+    s.erase(std::find_if(s.rbegin(), s.rend(), [](unsigned char ch) { return !std::isspace(ch); }).base(), s.end());
+    std::strncpy(dest, s.c_str(), destlen);
```
Comment (th1000s) [inline misc/pstring.cpp:55]: Since you are benchmarking, a `std::string_view` and `remove_prefix/suffix` (since C++17) should be faster :)

Comment (winterheart) [reply misc/pstring.cpp:55]:
```cpp
int CleanupStr(char *dest, const char *src, std::size_t destlen) {
  if ((src != nullptr) && (dest != nullptr)) {
    std::string_view str{src};
    auto it1 = std::find_if_not(str.begin(), str.end(), [](char ch) { return std::isspace(ch); });
    str = str.substr(it1 - str.begin());

    auto it2 = std::find_if_not(str.rbegin(), str.rend(), [](char ch) { return std::isspace(ch); });
    str = str.substr(0, str.rend() - it2);

    int size = str.size() < destlen ? str.size() : destlen;
    std::strncpy(dest, str.data(), size);
    dest[size] = '\0';
  }
```

Would be faster, but no so fast as C:

```c
int CleanupStr(char *dest, const char *src, size_t destlen) {
  if (destlen == 0)
    return 0;

  const char *end;
  size_t out_size;

  // Trim leading space
  while (isspace((unsigned char)*src))
    src++;

  // All white spaces
  if (*src == 0) {
    *dest = 0;
    return 0;
  }

  // Trim trailing space
  end = src + strlen(src) - 1;
  while (end > src && isspace((unsigned char)*end))
    end--;
  end++;

  out_size = (end - src) < destlen - 1 ? (end - src) : destlen - 1;

  memcpy(dest, src, out_size);
  dest[out_size] = 0;

  return 0;
}
```
(Not tested properly, just need careful check)

**Comment by winterheart** [comment]
Need to split first commit into removing and changing code ones.

File: misc/pstring.cpp [hunk 2]
```diff
@@ -38,140 +38,23 @@
...
-    } else if (ch > ' ' && err >= 1) {
-      end = i;
-    }
+//    This function strips all leading and trailing spaces, keeping internal spaces. This goes for tabs too.
+int CleanupStr(char *dest, const char *src, std::size_t destlen) {
+  if ((src != nullptr) && (dest != nullptr)) {
+    std::string_view s{src};
+    s.remove_prefix(std::count_if(s.begin(), s.end(), [](unsigned char ch) { return !std::isspace(ch); }));
+    s.remove_suffix(std::count_if(s.rbegin(), s.rend(), [](unsigned char ch) { return !std::isspace(ch); }));
+    std::strncpy(dest, s.data(), destlen);
```
Comment (winterheart) [inline misc/pstring.cpp:56]: This is dangerous move. `string_view.data()` never returns null-terminated string, so at dest's end may be garbage. You had to manually detect end of string and add \0 to end.

File: misc/pstring.cpp [hunk 3]
```diff
@@ -38,46 +38,36 @@
...
+      s.copy(dest, s.length());
+    } else if (s.back() == '\0' && s.length() > destlen) {
+      s.copy(dest, destlen - 1);
+      dest[destlen - 1] = '\0';
+    } else if (s.back() != '\0' && s.length() + 1 <= destlen) {
+      s.copy(dest, s.length());
+      dest[s.length()] = '\0';
+    } else if (s.back() != '\0' && s.length() + 1 > destlen) {
+      s.copy(dest, destlen - 1);
+      dest[destlen - 1] = '\0';
```
Comment (winterheart) [inline misc/pstring.cpp]: This looks bulky. Have you tried my implementation?

```cpp
    int size = str.size() < destlen ? str.size() : destlen;
    std::strncpy(dest, str.data(), size);
    dest[size] = '\0';
```

**Comment by winterheart** [comment]
Tested on Linux, thank you!

**Pull request merged**